public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,1013 @@
+  To see the form model, add the following line after the 
+  closing `form` tag in the `hero-detail.component.html`:
+
++makeExample('reactive-forms/ts/app/hero-detail-3.component.html', 'form-value-json','app/hero-detail.component.html')(format=".")
+
+:marked
+  The `heroForm.value` returns the _form model_.
+  Piping it through the `JsonPipe` renders the model as JSON in the browser:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/json-output.png" width="400px" alt="JSON output")
+
+:marked
+  The initial `name` property value is the empty string. 
+  Type into the _name_ input box and watch the keystokes appear in the JSON.
+
+
+
+:marked
+  Great! You have the basics of a form. 
+  
+  In real life apps, forms get big fast. 
+  `FormBuilder` makes form development and maintenance easier.
+
+
+.l-main-section
+a#formbuilder
+:marked
+  ## Introduction to _FormBuilder_
+
+  The `FormBuilder` class helps reduce repetition and 
+  clutter by handling details of control creation for you. 
+
+  To use `FormBuilder`, you need to import it into `hero-detail.component.ts`:
++makeExample('reactive-forms/ts/app/hero-detail-3a.component.ts', 'imports','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  Use it now to refactor the `HeroDetailComponent` into something that's a little easier to read and write,
+  by following this plan:
+
+  * Explicitly declare the type of the `heroForm` property to be `FormGroup`; you'll initialize it later.
+  * Inject a `FormBuilder` into the constructor.
+  * Add a new method that uses the `FormBuilder` to define the `heroForm`; call it `createForm`.
+  * Call `createForm` in the constructor.
+
+  The revised `HeroDetailComponent` looks like this:
++makeExample('reactive-forms/ts/app/hero-detail-3a.component.ts', 'v3a','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  `FormBuilder.group` is a factory method that creates a `FormGroup`. &nbsp;
+  `FormBuilder.group` takes an object whose keys and values are `FormControl` names and their definitions. 
+  In this example, the `name` control is defined by its initial data value, an empty string.
+
+  Defining a group of controls in a single object makes for a compact, readable style.
+  It beats writing an equivalent series of `new FormControl(...)` statements.
+
+a#validators
+:marked
+ ### Validators.required
+ Though this guide doesn't go deeply into validations, here is one example that 
+ demonstrates the simplicity of using `Validators.required` in reactive forms. 
+
+ First, be sure to import it:
++makeExample('reactive-forms/ts/app/hero-detail-3.component.ts', 'imports','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+ Then, you can easily make the `name` `FormControl` required by supplying the `name` 
+ property in the class with an array that holds two arguments, 
+ the initial value for `name` and `Validators.required`.  
```

Comment:
Glad we are including this, but there is no explanation of *why* reactive validators are simpler to use. We need to mention that in TD forms, you must wrap validators in a directive for comparison and that reactive validators are simple, composable functions. Otherwise it doesn't make sense why this is here.